lib/ansible/modules/cloud/amazon/dms_endpoint.py

```diff
@@ -0,0 +1,453 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: dms_endpoint
+short_description: creates or destroys a data migration services endpoint
+description:
+    - creates or destroys a data migration services endpoint,
+      that can be used to replicate data.
+version_added: "2.8"
+options:
+    state:
+      description:
+        - State of the endpoint
+      default: present
+      choices: ['present', 'absent']
+    endpointidentifier:
+      description:
+        - An identifier name for the endpoint
+    endpointtype:
+      description:
+        - Type of endpoint we want to manage
+      default: source
+      choices: ['source', 'target']
+    enginename:
+      description:
+        - Database engine that we want to use, please refer to
+          the AWS DMS for more information on the supported
+          engines and their limitation
+      choices: ['mysql', 'oracle', 'postgres', 'mariadb', 'aurora',
+                         'redshift', 's3', 'db2', 'azuredb', 'sybase',
+                         'dynamodb', 'mongodb', 'sqlserver']
+    username:
+      description:
+        - Username our endpoint will use to connect to the database
+    password:
+      description:
+        - Password used to connect to the database
+          this attribute can only be written
+          the AWS API does not return this parameter
+    servername:
+      description:
+        - Servername that the endpoint will connect to
+    port:
+      description:
+        - TCP port for access to the database
+    databasename:
+      description:
+        - Name for the database on the origin or target side
+    extraconnectionattributes:
+      description:
+        - Extra attributes for the database connection, the AWS documentation
+          states " For more information about extra connection attributes,
+          see the documentation section for your data store."
+    kmskeyid:
+      description:
+        - Encryption key to use to encrypt replication storage and
+          connection information
+    tags:
+      description:
+        - A list of tags to add to the endpoint
+    certificatearn:
+       description:
+         -  Amazon Resource Name (ARN) for the certificate
+    sslmode:
+       description:
+         - Mode used for the ssl connection
+       default: none
+       choices: ['none', 'require', 'verify-ca', 'verify-full']
+    serviceaccessrolearn:
+       description:
+         -  Amazon Resource Name (ARN) for the service access role that you
+            want to use to create the endpoint.
+    externaltabledefinition:
+       description:
+         - The external table definition
+    dynamodbsettings:
+       description:
+         - Settings in JSON format for the target Amazon DynamoDB endpoint
+           if source or target is dynamodb
+    s3settings:
+       description
```

Comment:
This is your problem, you're missing a colon. I had to load the whole thing using `yaml.load` to spot it

```
>>> DOCUMENTATION = '''...
>>> import yaml
>>> docs = yaml.load(DOCUMENTATION)
>>> for opt in docs['options']:
...     print opt, type(docs['options'][opt])
... 
username <type 'dict'>
elasticsearchsettings <type 'dict'>
servername <type 'dict'>
dynamodbsettings <type 'dict'>
serviceaccessrolearn <type 'dict'>
kmskeyid <type 'dict'>
dmstransfersettings <type 'dict'>
enginename <type 'dict'>
password <type 'dict'>
endpointidentifier <type 'dict'>
port <type 'dict'>
wait <type 'dict'>
certificatearn <type 'dict'>
tags <type 'dict'>
endpointtype <type 'dict'>
state <type 'dict'>
externaltabledefinition <type 'dict'>
extraconnectionattributes <type 'dict'>
databasename <type 'dict'>
sslmode <type 'dict'>
mongodbsettings <type 'dict'>
kinesissettings <type 'dict'>
s3settings <type 'str'>
```

Reply:
@willthames thank you for your help.